packages/material-ui/src/useWidth/unstable_useMediaQuery.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable camelcase, import/prefer-default-export */
+import React from 'react';
+import debounce from 'debounce'; // < 1kb payload overhead when lodash/debounce is > 3kb.
+import mediaQuery from 'css-mediaquery'; // Support SSR and old browsers
+
+const browser = typeof window !== 'undefined';
+
+function matches(query, valuesInput) {
+  if (browser && window.matchMedia) {
+    return window.matchMedia(query).matches;
+  }
+
+  let values = valuesInput;
+
+  if (browser && !values) {
+    values = {
+      width: window.innerWidth,
+      height: window.innerHeight,
+    };
+  }
+
+  return mediaQuery.match(query, values);
+}
+
+export function unstable_useMediaQuery(queryInput, options = {}) {
+  const {
+    noSSR = false,
+    resizeInterval = 166, // Corresponds to 10 frames at 60 Hz.
+    values,
+  } = options;
+
+  const query = queryInput.replace('@media ', '');
+  const state = noSSR ? matches(query, values) : false;
+  const [match, setMatch] = React.useState(state);
+
+  if (browser) {
```

Comment:
I'm not comfortable having this in the codebase. It's technically correct since the call order stays the same for a given environment but hooks are new and until they are stable (in this case I'm more concerned about the lint rules) and battle tested we should follow the rules. This means moving this into the callback of the `useEffect.`

Reply:
I don't have any opinions about hooks usage (yet). I just do what my linter tells me to ;)

Reply:
Happy to add `eslint-plugin-react-hooks` in the project!

Reply:
I think adding `eslint-plugin-react-hooks` is a good idea

Reply:
Moving the check to the second argument of useEffect is the correct decision :+1: